### PR TITLE
Added support for Symfony3

### PR DESCRIPTION
**Pull request by pkruithof**
Note that I could not test this myself due to the composer autoload dump not working:

```
$ composer dump
You are running composer with xdebug enabled. This has a major impact on runtime performance. See https://getcomposer.org/xdebug
Generating autoload files
Warning: Could not load Puli configuration: InvalidConfigException: The configuration in /Users/peter/projects/puli-symfony-bundle/puli.json is  
  invalid:                                                                     
  packages.doctrine/instantiator: The property env is not defined and the      
  definition does not allow additional properties
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,6 +32,7 @@ before_install:
   - travis_retry composer self-update && composer --version
 
 install:
+  - if [ "$SYMFONY_VERSION" != "" ]; then travis_retry composer require "symfony/symfony:${SYMFONY_VERSION}" --prefer-source --no-update -n; fi;
```
Comment (webmozart) [inline .travis.yml]: Do I understand correctly that the next line is executed nevertheless? Doesn't the next line undo the changes here?

Comment (pkruithof) [inline .travis.yml]: No, the `require` writes the change to the `composer.json` file.

See also: https://docs.travis-ci.com/user/languages/php/#Testing-Against-Multiple-Versions-of-Dependencies-(e.g.-Symfony)

**Comment by pkruithof** [comment]
Could you tell me how to solve the `env` property issue, so I can test this change properly?

**Comment by webmozart** [comment]
That issue is fixed here: https://github.com/puli/manager/pull/30

**Comment by pkruithof** [comment]
The [failing 2.7 build](https://travis-ci.org/puli/symfony-bundle/jobs/97444118) looks like a Symfony issue, the failing 3.0 build is due to a package requirement in the composer plugin ([PR](https://github.com/puli/composer-plugin/pull/29))

**Comment by webmozart** [comment]
What's the status here?

**Comment by Spomky** [comment]
Any news regarding this PR?

**Comment by webmozart** [comment]
I restarted the Travis build, let's see what happens.

File: .travis.yml (hunk 1 above)
Comment (webmozart) [inline .travis.yml]: From the bundle's composer.json:

```json
    "require": {
        "symfony/framework-bundle": "^2.7.3",
        "symfony/dependency-injection": "^2.7.3",
        "puli/composer-plugin": "^1.0-beta8",
        "puli/symfony-bridge": "^1.0-beta3"
    },
    "require-dev": {
        "symfony/twig-bundle": "^2.7.3",
        "puli/twig-extension": "^1.0-beta7",
        "puli/discovery": "^1.0-beta8",
        "puli/cli": "^1.0-beta8",
        "phpunit/phpunit": "^4.6",
        "sebastian/version": "^1.0.1"
    },
```

Hence this line should be:

```
  - if [ "$SYMFONY_VERSION" != "" ]; then travis_retry composer require "symfony/framework-bundle:${SYMFONY_VERSION}" "symfony/dependency-injection:${SYMFONY_VERSION}" --no-update -n; travis_retry composer require "symfony/twig-bundle:${SYMFONY_VERSION}" --no-update -n; fi;
```

As far as I understand the --prefer-source can be removed since we're doing --no-update anyway.

@pkruithof could you update this PR?

**Comment by webmozart** [comment]
Apart from my last code comment, this is only waiting for https://github.com/padraic/phar-updater/pull/16 to be complete.

**Comment by pkruithof** [comment]
I think you also need to tag a new puli/cli release:

```
puli/cli 1.0.0-beta9 requires symfony/filesystem ^2.3
```

**Comment by webmozart** [comment]
This is caused by the issue in phar-updater that I just mentioned.

File: .travis.yml [hunk 2]
```diff
@@ -26,6 +32,8 @@ before_install:
   - travis_retry composer self-update && composer --version
 
 install:
+  - if [ "$SYMFONY_VERSION" != "" ]; then travis_retry composer require "symfony/framework-bundle:${SYMFONY_VERSION}" "symfony/dependency-injection:${SYMFONY_VERSION}" --no-update -n; travis_retry composer require "symfony/twig-bundle:${SYMFONY_VERSION}" --no-update -n; fi;
+  - if [ "$SYMFONY_VERSION" != "" ]; then travis_retry composer require "symfony/symfony:${SYMFONY_VERSION}" --prefer-source --no-update -n; fi;
```
Comment (webmozart) [inline .travis.yml]: This line should be removed

Comment (webmozart) [inline .travis.yml]: I also think you can remove the `travis_retry` parts.

**Comment by pkruithof** [comment]
The phar-updater is the reason for the finder requirement, but cli had a filesystem requirement that was updated since beta9:

https://github.com/puli/cli/blob/1.0.0-beta9/composer.json#L23

File: composer.json [hunk 3]
```diff
@@ -10,13 +10,13 @@
         }
     ],
     "require": {
-        "symfony/framework-bundle": "^2.7.3",
-        "symfony/dependency-injection": "^2.7.3",
+        "symfony/framework-bundle": "^2.7.3|^3.0.0",
```
Comment (webmozart) [inline composer.json]: `^3.0` should be sufficient

**Comment by webmozart** [comment]
Thanks @pkruithof! :)

**Comment by webmozart** [comment]
> The phar-updater is the reason for the finder requirement, but cli had a filesystem requirement that was updated since beta9:

Yes, but with "minimum-stability" set to "dev" this shouldn't be a problem. :)

File: .travis.yml (hunk 2 above)
Comment (pkruithof) [inline .travis.yml]: Woops!

I'm hesitant about the `travis_retry` here as well. This was copied from other examples, but I'll just remove it here for now.

**Comment by pkruithof** [comment]
Not sure why appveyor is failing, everything else seems to pass...

**Comment by webmozart** [comment]
It was trickier than I expected! Fixed here: https://github.com/puli/symfony-bundle/pull/23

**Comment by Spomky** [comment]
Good to know that.
Do you plan to add a tag soon?

**Comment by webmozart** [comment]
Yes, about mid January or earlier.
